frontend/lib/src/AppNode.ts

```diff
@@ -142,16 +147,21 @@ export class ElementNode implements AppNode {
 
   private lazyVegaLiteChartElement?: VegaLiteChartElement
 
+  // The hash of the script that created this element.
+  public readonly activeScriptHash: string
+
   /** Create a new ElementNode. */
   public constructor(
     element: Element,
     metadata: ForwardMsgMetadata,
     scriptRunId: string,
+    activeScriptHash: string,
```

Comment:
I think we'll want to double-check that `ElementNode`s aren't constructed by notebooks so that we can either make this optional or given the notebooks team that an upgrade to v1.35 will require a bit of work on their end